components/style/properties/properties.mako.rs

```diff
@@ -2665,6 +2663,14 @@ pub fn apply_declarations<'a, F, I>(device: &Device,
                                                  error_reporter);
             % endif
             }
+
+            let writing_mode = get_writing_mode(context.style.get_inheritedbox());
+            context.style.writing_mode = writing_mode;
```

Comment:
Checking the spec, I believe `writing-mode` actually should be set before `font-size`, because font size may depend on that property when any font-relative length unit is used on it.